### ISSUE TITLE
Unarchive module not working with "mode" option set and tar.gz archives 

### ISSUE THREAD

**Issue by ansibot** [ansible/ansible#29533]
From @alxnov on 2016-10-31T18:50:29Z
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
unarchive

##### ANSIBLE VERSION
```
ansible-playbook 2.1.1.0
  config file = 
  configured module search path = Default w/o overrides
```

##### CONFIGURATION
N/A

##### OS / ENVIRONMENT
N/A

##### SUMMARY
Unarchive does not work with `mode:` options set for a tar.gz archive, seems to work fine for a sample zip archive.

##### STEPS TO REPRODUCE
```
✗ cat > inv3.yml <<EOF      
[servicetest]
${MYSERVERIP}

[servicetest:vars]
ansible_ssh_user=${MYUSER}
ansible_ssh_private_key_file=${MYSSHPEM}
EOF

✗ tar -cvz -f foo.tar.gz inv3.yml                                              
a inv3.yml

✗ scp -i ${MYSSHPEM} foo.tar.gz ${MYUSER}@${MYSERVERIP}:~/foo.tar.gz
foo.tar.gz

✗ cat baz.yml
- hosts: servicetest
  tasks:
    - unarchive:
        src: /home/ec2-user/foo.tar.gz
        dest: /home/ec2-user/foo
        copy: no
        mode: 0750

✗ ansible-playbook -i inv3.yml baz.yml -vvvv
No config file found; using defaults
Loaded callback default of type stdout, v2.0

PLAYBOOK: baz.yml **************************************************************
1 plays in baz.yml

PLAY [servicetest] *************************************************************

TASK [setup] *******************************************************************
<10.11.12.13> ESTABLISH SSH CONNECTION FOR USER: ansible-user
<10.11.12.13> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o 'IdentityFile="/Users/alxnov/.ssh/ansible-user.pem"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ansible-user -o ConnectTimeout=10 -o ControlPath=/Users/alxnov/.ansible/cp/ansible-ssh-%h-%p-%r 10.11.12.13 '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1477938800.66-23242196276095 `" && echo ansible-tmp-1477938800.66-23242196276095="` echo $HOME/.ansible/tmp/ansible-tmp-1477938800.66-23242196276095 `" ) && sleep 0'"'"''
<10.11.12.13> PUT /var/folders/rs/rf66zygj1tqgx6whbtstr38w0000gp/T/tmpirT5Ap TO /home/ansible-user/.ansible/tmp/ansible-tmp-1477938800.66-23242196276095/setup
<10.11.12.13> SSH: EXEC sftp -b - -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o 'IdentityFile="/Users/alxnov/.ssh/ansible-user.pem"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ansible-user -o ConnectTimeout=10 -o ControlPath=/Users/alxnov/.ansible/cp/ansible-ssh-%h-%p-%r '[10.11.12.13]'
<10.11.12.13> ESTABLISH SSH CONNECTION FOR USER: ansible-user
<10.11.12.13> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o 'IdentityFile="/Users/alxnov/.ssh/ansible-user.pem"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ansible-user -o ConnectTimeout=10 -o ControlPath=/Users/alxnov/.ansible/cp/ansible-ssh-%h-%p-%r -tt 10.11.12.13 '/bin/sh -c '"'"'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /home/ansible-user/.ansible/tmp/ansible-tmp-1477938800.66-23242196276095/setup; rm -rf "/home/ansible-user/.ansible/tmp/ansible-tmp-1477938800.66-23242196276095/" > /dev/null 2>&1 && sleep 0'"'"''
ok: [10.11.12.13]

TASK [unarchive] ***************************************************************
task path: /Users/alxnov/Workspace/firewalkwithme/playbooks/fluffy/baz.yml:3
<10.11.12.13> ESTABLISH SSH CONNECTION FOR USER: ansible-user
<10.11.12.13> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o 'IdentityFile="/Users/alxnov/.ssh/ansible-user.pem"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ansible-user -o ConnectTimeout=10 -o ControlPath=/Users/alxnov/.ansible/cp/ansible-ssh-%h-%p-%r 10.11.12.13 '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1477938801.5-158057866662852 `" && echo ansible-tmp-1477938801.5-158057866662852="` echo $HOME/.ansible/tmp/ansible-tmp-1477938801.5-158057866662852 `" ) && sleep 0'"'"''
<10.11.12.13> ESTABLISH SSH CONNECTION FOR USER: ansible-user
<10.11.12.13> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o 'IdentityFile="/Users/alxnov/.ssh/ansible-user.pem"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ansible-user -o ConnectTimeout=10 -o ControlPath=/Users/alxnov/.ansible/cp/ansible-ssh-%h-%p-%r 10.11.12.13 '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1477938801.61-150668743544373 `" && echo ansible-tmp-1477938801.61-150668743544373="` echo $HOME/.ansible/tmp/ansible-tmp-1477938801.61-150668743544373 `" ) && sleep 0'"'"''
<10.11.12.13> PUT /var/folders/rs/rf66zygj1tqgx6whbtstr38w0000gp/T/tmpvDaqWq TO /home/ansible-user/.ansible/tmp/ansible-tmp-1477938801.61-150668743544373/stat
<10.11.12.13> SSH: EXEC sftp -b - -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o 'IdentityFile="/Users/alxnov/.ssh/ansible-user.pem"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ansible-user -o ConnectTimeout=10 -o ControlPath=/Users/alxnov/.ansible/cp/ansible-ssh-%h-%p-%r '[10.11.12.13]'
<10.11.12.13> ESTABLISH SSH CONNECTION FOR USER: ansible-user
<10.11.12.13> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o 'IdentityFile="/Users/alxnov/.ssh/ansible-user.pem"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ansible-user -o ConnectTimeout=10 -o ControlPath=/Users/alxnov/.ansible/cp/ansible-ssh-%h-%p-%r -tt 10.11.12.13 '/bin/sh -c '"'"'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /home/ansible-user/.ansible/tmp/ansible-tmp-1477938801.61-150668743544373/stat; rm -rf "/home/ansible-user/.ansible/tmp/ansible-tmp-1477938801.61-150668743544373/" > /dev/null 2>&1 && sleep 0'"'"''
<10.11.12.13> ESTABLISH SSH CONNECTION FOR USER: ansible-user
<10.11.12.13> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o 'IdentityFile="/Users/alxnov/.ssh/ansible-user.pem"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ansible-user -o ConnectTimeout=10 -o ControlPath=/Users/alxnov/.ansible/cp/ansible-ssh-%h-%p-%r 10.11.12.13 '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1477938802.07-258573422285916 `" && echo ansible-tmp-1477938802.07-258573422285916="` echo $HOME/.ansible/tmp/ansible-tmp-1477938802.07-258573422285916 `" ) && sleep 0'"'"''
<10.11.12.13> PUT /var/folders/rs/rf66zygj1tqgx6whbtstr38w0000gp/T/tmpjZnZzw TO /home/ansible-user/.ansible/tmp/ansible-tmp-1477938802.07-258573422285916/unarchive
<10.11.12.13> SSH: EXEC sftp -b - -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o 'IdentityFile="/Users/alxnov/.ssh/ansible-user.pem"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ansible-user -o ConnectTimeout=10 -o ControlPath=/Users/alxnov/.ansible/cp/ansible-ssh-%h-%p-%r '[10.11.12.13]'
<10.11.12.13> ESTABLISH SSH CONNECTION FOR USER: ansible-user
<10.11.12.13> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o 'IdentityFile="/Users/alxnov/.ssh/ansible-user.pem"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ansible-user -o ConnectTimeout=10 -o ControlPath=/Users/alxnov/.ansible/cp/ansible-ssh-%h-%p-%r -tt 10.11.12.13 '/bin/sh -c '"'"'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /home/ansible-user/.ansible/tmp/ansible-tmp-1477938802.07-258573422285916/unarchive; rm -rf "/home/ansible-user/.ansible/tmp/ansible-tmp-1477938802.07-258573422285916/" > /dev/null 2>&1 && sleep 0'"'"''
<10.11.12.13> ESTABLISH SSH CONNECTION FOR USER: ansible-user
<10.11.12.13> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o 'IdentityFile="/Users/alxnov/.ssh/ansible-user.pem"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ansible-user -o ConnectTimeout=10 -o ControlPath=/Users/alxnov/.ansible/cp/ansible-ssh-%h-%p-%r 10.11.12.13 '/bin/sh -c '"'"'rm -f -r /home/ansible-user/.ansible/tmp/ansible-tmp-1477938801.5-158057866662852/ > /dev/null 2>&1 && sleep 0'"'"''
fatal: [10.11.12.13]: FAILED! => {"changed": false, "failed": true, "invocation": {"module_args": {"backup": null, "content": null, "copy": false, "creates": null, "delimiter": null, "dest": "/home/ansible-user/foo", "directory_mode": null, "exclude": [], "extra_opts": [], "follow": false, "force": null, "group": null, "keep_newer": false, "list_files": false, "mode": 488, "original_basename": "foo.tar.gz", "owner": null, "regexp": null, "remote_src": null, "selevel": null, "serole": null, "setype": null, "seuser": null, "src": "/home/ansible-user/foo.tar.gz"}}, "msg": "path /home/ansible-user/foo/inv3.yml does not exist", "path": "/home/ansible-user/foo/inv3.yml", "state": "absent"}

NO MORE HOSTS LEFT *************************************************************
 [WARNING]: Could not create retry file 'baz.retry'.         [Errno 2] No such file or directory: ''


PLAY RECAP *********************************************************************
10.11.12.13                : ok=1    changed=0    unreachable=0    failed=1
```

##### EXPECTED RESULTS
```
✗ grep -v mode baz.yml > baz2.yml

✗ ansible-playbook -i inv3.yml baz2.yml -vvvv
No config file found; using defaults
Loaded callback default of type stdout, v2.0

PLAYBOOK: baz2.yml *************************************************************
1 plays in baz2.yml

PLAY [servicetest] *************************************************************

TASK [setup] *******************************************************************
<10.11.12.13> ESTABLISH SSH CONNECTION FOR USER: ansible-user
<10.11.12.13> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o 'IdentityFile="/Users/alxnov/.ssh/ansible-user.pem"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ansible-user -o ConnectTimeout=10 -o ControlPath=/Users/alxnov/.ansible/cp/ansible-ssh-%h-%p-%r 10.11.12.13 '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1477939280.81-250171781459807 `" && echo ansible-tmp-1477939280.81-250171781459807="` echo $HOME/.ansible/tmp/ansible-tmp-1477939280.81-250171781459807 `" ) && sleep 0'"'"''
<10.11.12.13> PUT /var/folders/rs/rf66zygj1tqgx6whbtstr38w0000gp/T/tmpqYQRYi TO /home/ansible-user/.ansible/tmp/ansible-tmp-1477939280.81-250171781459807/setup
<10.11.12.13> SSH: EXEC sftp -b - -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o 'IdentityFile="/Users/alxnov/.ssh/ansible-user.pem"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ansible-user -o ConnectTimeout=10 -o ControlPath=/Users/alxnov/.ansible/cp/ansible-ssh-%h-%p-%r '[10.11.12.13]'
<10.11.12.13> ESTABLISH SSH CONNECTION FOR USER: ansible-user
<10.11.12.13> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o 'IdentityFile="/Users/alxnov/.ssh/ansible-user.pem"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ansible-user -o ConnectTimeout=10 -o ControlPath=/Users/alxnov/.ansible/cp/ansible-ssh-%h-%p-%r -tt 10.11.12.13 '/bin/sh -c '"'"'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /home/ansible-user/.ansible/tmp/ansible-tmp-1477939280.81-250171781459807/setup; rm -rf "/home/ansible-user/.ansible/tmp/ansible-tmp-1477939280.81-250171781459807/" > /dev/null 2>&1 && sleep 0'"'"''
ok: [10.11.12.13]

TASK [unarchive] ***************************************************************
task path: /Users/alxnov/Workspace/firewalkwithme/playbooks/fluffy/baz2.yml:3
<10.11.12.13> ESTABLISH SSH CONNECTION FOR USER: ansible-user
<10.11.12.13> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o 'IdentityFile="/Users/alxnov/.ssh/ansible-user.pem"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ansible-user -o ConnectTimeout=10 -o ControlPath=/Users/alxnov/.ansible/cp/ansible-ssh-%h-%p-%r 10.11.12.13 '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1477939281.62-278860514056026 `" && echo ansible-tmp-1477939281.62-278860514056026="` echo $HOME/.ansible/tmp/ansible-tmp-1477939281.62-278860514056026 `" ) && sleep 0'"'"''
<10.11.12.13> ESTABLISH SSH CONNECTION FOR USER: ansible-user
<10.11.12.13> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o 'IdentityFile="/Users/alxnov/.ssh/ansible-user.pem"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ansible-user -o ConnectTimeout=10 -o ControlPath=/Users/alxnov/.ansible/cp/ansible-ssh-%h-%p-%r 10.11.12.13 '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1477939281.73-224375086385264 `" && echo ansible-tmp-1477939281.73-224375086385264="` echo $HOME/.ansible/tmp/ansible-tmp-1477939281.73-224375086385264 `" ) && sleep 0'"'"''
<10.11.12.13> PUT /var/folders/rs/rf66zygj1tqgx6whbtstr38w0000gp/T/tmptlnx43 TO /home/ansible-user/.ansible/tmp/ansible-tmp-1477939281.73-224375086385264/stat
<10.11.12.13> SSH: EXEC sftp -b - -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o 'IdentityFile="/Users/alxnov/.ssh/ansible-user.pem"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ansible-user -o ConnectTimeout=10 -o ControlPath=/Users/alxnov/.ansible/cp/ansible-ssh-%h-%p-%r '[10.11.12.13]'
<10.11.12.13> ESTABLISH SSH CONNECTION FOR USER: ansible-user
<10.11.12.13> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o 'IdentityFile="/Users/alxnov/.ssh/ansible-user.pem"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ansible-user -o ConnectTimeout=10 -o ControlPath=/Users/alxnov/.ansible/cp/ansible-ssh-%h-%p-%r -tt 10.11.12.13 '/bin/sh -c '"'"'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /home/ansible-user/.ansible/tmp/ansible-tmp-1477939281.73-224375086385264/stat; rm -rf "/home/ansible-user/.ansible/tmp/ansible-tmp-1477939281.73-224375086385264/" > /dev/null 2>&1 && sleep 0'"'"''
<10.11.12.13> ESTABLISH SSH CONNECTION FOR USER: ansible-user
<10.11.12.13> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o 'IdentityFile="/Users/alxnov/.ssh/ansible-user.pem"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ansible-user -o ConnectTimeout=10 -o ControlPath=/Users/alxnov/.ansible/cp/ansible-ssh-%h-%p-%r 10.11.12.13 '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1477939282.19-225224644675241 `" && echo ansible-tmp-1477939282.19-225224644675241="` echo $HOME/.ansible/tmp/ansible-tmp-1477939282.19-225224644675241 `" ) && sleep 0'"'"''
<10.11.12.13> PUT /var/folders/rs/rf66zygj1tqgx6whbtstr38w0000gp/T/tmpsMHgOg TO /home/ansible-user/.ansible/tmp/ansible-tmp-1477939282.19-225224644675241/unarchive
<10.11.12.13> SSH: EXEC sftp -b - -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o 'IdentityFile="/Users/alxnov/.ssh/ansible-user.pem"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ansible-user -o ConnectTimeout=10 -o ControlPath=/Users/alxnov/.ansible/cp/ansible-ssh-%h-%p-%r '[10.11.12.13]'
<10.11.12.13> ESTABLISH SSH CONNECTION FOR USER: ansible-user
<10.11.12.13> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o 'IdentityFile="/Users/alxnov/.ssh/ansible-user.pem"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ansible-user -o ConnectTimeout=10 -o ControlPath=/Users/alxnov/.ansible/cp/ansible-ssh-%h-%p-%r -tt 10.11.12.13 '/bin/sh -c '"'"'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /home/ansible-user/.ansible/tmp/ansible-tmp-1477939282.19-225224644675241/unarchive; rm -rf "/home/ansible-user/.ansible/tmp/ansible-tmp-1477939282.19-225224644675241/" > /dev/null 2>&1 && sleep 0'"'"''
<10.11.12.13> ESTABLISH SSH CONNECTION FOR USER: ansible-user
<10.11.12.13> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o 'IdentityFile="/Users/alxnov/.ssh/ansible-user.pem"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ansible-user -o ConnectTimeout=10 -o ControlPath=/Users/alxnov/.ansible/cp/ansible-ssh-%h-%p-%r 10.11.12.13 '/bin/sh -c '"'"'rm -f -r /home/ansible-user/.ansible/tmp/ansible-tmp-1477939281.62-278860514056026/ > /dev/null 2>&1 && sleep 0'"'"''
changed: [10.11.12.13] => {"changed": true, "dest": "/home/ansible-user/foo", "extract_results": {"cmd": "/usr/bin/gtar -C \"/home/ansible-user/foo\" -xz -f \"/home/ansible-user/foo.tar.gz\"", "err": "", "out": "", "rc": 0}, "gid": 1000, "group": "ansible-user", "handler": "TgzArchive", "invocation": {"module_args": {"backup": null, "content": null, "copy": false, "creates": null, "delimiter": null, "dest": "/home/ansible-user/foo", "directory_mode": null, "exclude": [], "extra_opts": [], "follow": false, "force": null, "group": null, "keep_newer": false, "list_files": false, "mode": null, "original_basename": "foo.tar.gz", "owner": null, "regexp": null, "remote_src": null, "selevel": null, "serole": null, "setype": null, "seuser": null, "src": "/home/ansible-user/foo.tar.gz"}}, "mode": "0775", "owner": "ansible-user", "secontext": "unconfined_u:object_r:user_home_t:s0", "size": 4096, "src": "/home/ansible-user/foo.tar.gz", "state": "directory", "uid": 1000}

PLAY RECAP *********************************************************************
10.11.12.13                : ok=2    changed=1    unreachable=0    failed=0   
```
Copied from original issue: ansible/ansible-modules-core#5445

**Comment by ansibot** [comment]
From @ansibot on 2016-10-31T18:50:29Z



@dagwieers, @pileofrogs, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
From @dagwieers on 2016-10-31T18:50:29Z

Please try with the latest unarchive from Ansible 2.2.0 or the devel branch.

**Comment by GhostPratt** [comment]
@alxnov I was unable to reproduce this issue with the current development branch.  Could you retest this?  If not we'll close this as fixed on development.  Thanks.

**Comment by abadger** [comment]
needs_info

**Comment by ansibot** [comment]
@alxnov This issue is waiting for your response. Please respond or the issue will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@alxnov You have not responded to information requests in this issue so we will assume it no longer affects you. If you are still interested in this, please create a new issue with the requested information.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by krasowskir** [comment]
it still doesn't work. 

ansible --version
ansible 2.9.0.dev0
  config file = None
  python version = 2.7.13 (default, Sep 26 2018, 18:42:22) [GCC 6.3.0 20170516]


my playbook:
```yml- hosts: home 
  become: yes
  become_method: sudo
  tasks: 
  - name: download maven is at the latest version 
    get_url: 
     url: http://apache.mirror.iphh.net/maven/maven-3/3.6.1/binaries/apache-maven-3.6.1-bin.tar.gz
     dest: /home/richard/Programme/
     mode: 0644
     group: richard
     owner: richard
  - name: unpack apache
    unarchive:
     src: /home/richard/Programme/apache-maven-3.6.1-bin.tar.gz
     dest: /home/richard/Programme/apache-maven-3.6.1
     mode: 0644
     group: richard
     owner: richard
  - name: Echo path
    shell: echo $PATH  
    environment:
     PATH: $PATH:/home/richard/Programme/apache-maven-3.6.1/bin
  - name: add PATH to bashrc
    shell: echo "export PATH=$PATH:/home/richard/Programme/apache-maven-3.6.1/bin" >> /home/richard/.bashrc

**Comment by krasowskir** [comment]
[result.txt](https://github.com/ansible/ansible/files/3219372/result.txt)

**Comment by krasowskir** [comment]
```bash 
drwxr-xr-x  7 root    root       4096 Mai 25 13:40 apache-maven-3.6.1
-rw-r--r--  1 richard richard 9136463 Mai 25 13:35 apache-maven-3.6.1-bin.tar.gz
```
